<launch>
    <arg name="quad_name" default="kingfisher"/>
    <!-- <arg name="quad_name" default="iris"/> -->


    <!-- Start Octomap Server -->
    

    <node pkg="octomap_server" type="octomap_server_node" name="octomap_server">
        <param name="resolution" value="0.1"/>
        <param name="frame_id" value="world"/>     
        <param name="base_frame_id" value="/$(arg quad_name)/base_link"/> 
        <!-- BELOW TRUE< CHECK FALSE (SUPPOSEDLY BETTTER FORLARGE MAPS -->
        <param name="latch" value="false"/>  
        <param name="track_changes" value="true"/> 
        <param name="compress_map" value="false"/>
        <param name="incremental_2D_projection" value="true"/>
        <param name="publish_free_space" value="true"/>
        <!-- <param name="publish_frequency" value="1.0"/>  -->
        <param name="sensor_model/max_range" value="5.0"/>
        <remap from="/cloud_in" to="/$(arg quad_name)/camera_depth/depth/points"/>
    </node>














    <!-- Start Frontier Detector -->
    <node pkg="autonomous_exploration" type="frontier_detector" name="frontier_detector" output="screen"/>

`
    <node pkg="autonomous_exploration" type="waypoint_publisher" name="waypoint_publisher" output="screen"/>

    <!-- Start trajectory publishing from ego to agilicious -->
    <node pkg="autonomous_exploration" type="trajectory_publisher" name="trajectory_publisher" output="screen"/>


    <!-- Track performance -->
    <node pkg="autonomous_exploration" type="performance_tracker" name="performance_tracker" output="screen"/>


    <!-- Track entropy -->
    <node pkg="autonomous_exploration" type="entropy_calculator" name="entropy_calculator" output="screen"/>
 

    <!-- Log energy -->
    <node pkg="autonomous_exploration" type="energy_logger" name="energy_logger" output="screen"/>
 


</launch>
